modules/userdirectory/src/main/java/org/opencastproject/userdirectory/JpaGroupRoleProvider.java

```diff
@@ -275,6 +303,10 @@ public void updateGroupMembershipFromRoles(String userName, String orgId, List<S
     // List of the user's groups
     List<JpaGroup> membership = UserDirectoryPersistenceUtil.findGroupsByUser(userName, orgId, emf);
     for (JpaGroup group : membership) {
+      if (prefix != null && !"".equals(prefix) && !group.getRole().startsWith(prefix)) {
```

Comment:
Maybe something like:
```suggestion
      if (StringUtils.isNotBlank(prefix) && !group.getRole().startsWith(prefix)) {
```